pkg/client/clientset/versioned/typed/jenkins.io/v1/app_expansion.go

```diff
@@ -0,0 +1,31 @@
+package v1
+
+import (
+	"github.com/jenkins-x/jx/pkg/apis/jenkins.io/v1"
+	util "github.com/jenkins-x/jx/pkg/util/json"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// AppExpansion expands the default CRUD interface for App.
+type AppExpansion interface {
+	PatchUpdate(app *v1.App) (result *v1.App, err error)
+}
+
+// PatchUpdate takes the representation of an app and updates using Patch generating a JSON patch to do so.
+// Returns the server's representation of the app, and an error, if there is any.
+func (c *apps) PatchUpdate(app *v1.App) (*v1.App, error) {
```

Comment:
@pmuir, So this is the essence of using JSON patch. Each CRD type gets its `PatchUpdate`. I am leveraging the go-client cache here, so I don't think we need the explicit deep clone.

Reply:
where is the generator for this?